python/ngen_cal/src/ngen/cal/ngen.py

```diff
@@ -212,6 +230,41 @@ def check_for_partitions(cls, values: dict):
             raise ValueError("Must provide partitions if using parallel")
         return values
 
+    @root_validator(pre=True)
```

Comment:
I don't think this needs to be `pre` does it?

Reply:
I wanted as early stop as possible, without hydro fabric the rest is kinda meaningless (at least for doing an actual calibration...).  It could probably be done effectively at any level...I was also just following in kind with the previous root validator...

Reply:
It looks like you can just remove the `pre=True` parameter to accomplish the same thing. With `pre=True` the validator will be the first to handle the input and it seems like what you really want is that `pydantic` does its coercion / validation, then you apply these invariants. The `root_validator` will still run during object initialization, just after `pydantic` has done coercion / validation.

Reply:
addressed in [8d60cb2](https://github.com/NOAA-OWP/ngen-cal/pull/128/commits/8d60cb23b31591671ebff9e3f5125435cc240aa6)